docs/instrumentation-libraries.md

```diff
@@ -0,0 +1,99 @@
+# Support for OpenTelemetry Instrumentation Libraries
+
+OpenTelemetry instrumentation are implemented using a variety of techniques.
+Sometimes the instrumentation is built directly into the library being
+instrumented, sometimes the instrumentation requires the addition of a separate
+instrumentation library, and other times it is a hybrid of those approaches. This
+project aims to make using these instrumentation approaches easier and more
+automatic for our users. This requires us to both vet and dynamically enable a
+subset of the instrumentation that is available to the OpenTelemetry community,
+while still providing some flexibility to allow users to manually load and enable
+instrumentation that we do not support out of the box.
+
+## Libraries and our instrumentation support plans
+
+The libraries listed here contain our current thoughts about which libraries should
+include out of the box support, and which libraries we decided to not support. If
+you have a library that you would like to consider adding to this list, please
+submit an issue to request its inclusion.
+
+### Application Frameworks
+
+1. ASP.NET
+1. AspNetCore
+1. Windows Workflow Foundation
+1. OWIN
+1. Azure Service Fabric
+
+### Databases
+
+#### General database frameworks
+
+1. ADO.NET
+1. EntityFramework
+1. Dapper
+1. System.Data.LINQ
+1. Microsoft.Practices.EnterpriseLibrary.Data
+1. EntityFrameworkCore
+
+#### Specific Databases
+
+1. MySQL
+    * MySql.data
+    * MySqlConnector
+1. PostgreSQL
+1. SQLite
+1. MongoDB
+1. MS SQL Server
+1. Redis
+    * StackExchange.Redis
+1. MariaDB
+1. Firebase
+1. Elasticsearch
+1. Oracle
+1. DynamoDB
+1. Cassandra
+1. IBM DB2
+1. Couchbase
+
+### External request libraries (REST/RPC)
```

Comment:
If we rename it to IPC, should the message queuing libraries be included there too?

Reply:
IPC is so broad that I would rather break this down into the following categories:

1. Http Clients
1. Messaging
1. RPC

Reply:
WCF can be used for REST, but I consider it more of a RPC framework. Should it be listed under RPC? I'm also wondering if grpc and WCF should be listed under frameworks because of the potential for running them in a self-hosted fashion instead of relying on ASP.NET or AspNetCore?

Reply:
Yes I consider both WCF and GRPC and RPC frameworks and consider them separately from ASP.NET and AspNetCore

Reply:
How about "IPC" and putting the messaging libraries here as well?